docs/hopsml/kfserving.rst

```diff
@@ -0,0 +1,243 @@
+The transformer script has to implement the `Transformer` class as well as the methods `preprocess` and `postprocess`. Both methods receive a dictionary containing the model input or predictions, and must also return a dictionary with the transformed values.
+
+.. code-block:: python
+
+    class Transformer(object):
+        def __init__(self):
+            print("[Transformer] Initializing...")
+            # Initialization code goes here
+
+        def preprocess(self, inputs):
+            # Transform the request inputs here. The object returned by this method will be used as model input.
+            return inputs
+
+        def postprocess(self, outputs):
+            # Transform the predictions computed by the model before returning a response.
+            return outputs
+
+The transformer script should be placed in the Models dataset in your Hopsworks project.
+You can add the script file by clicking on the "Browse" button next to *Transformer*. This will open a popup window that will allow you to browse your project and select the file.
+ 
+.. _kfserving_serving2.png: ../_images/serving/kfserving_serving2.png
+.. figure:: ../imgs/serving/kfserving_serving2.png
+    :alt: Select transformer script
+    :target: `kfserving_serving2.png`_
+    :align: center
+    :figclass: align-center
+
+Transformers use the environment of the project to which they belong, as well as jupyter notebooks and jobs.
+Therefore, you can install additional python libraries using the :ref:`python_environment` service and test your transformer code beforehand using Jupyter to make sure it will run correctly once it is deployed.
+
+Moreover, transformers scale independently and can make use of the CPU or GPU regardless how model servers are configured since they are deployed in a different Kubernetes pod.
+One of the potential use cases of transformers is to solve the Training/Serving skew problem by applying the same transformations to the model inputs than those used to generate the training dataset.
+Furthermore, they can be used to provide context to the model by enriching model inputs with the most recent feature values retrieved from the Online Feature Store using `HSFS <https://docs.hopsworks.ai/latest/>`_.
+
+When a transformer script is selected, the artifact version is set to "CREATE". Each time a service is created containing a transformer, a new artifact is generated with an incremental version in a folder named `Artifacts` under the model version folder in `Models` dataset.
+An artifact can be seen as a package containing all the necessary files to deploy a model (e.g model files, environment, transformer script,...).
+When only the model is served, that is, no transformer component is being deployed, the artifact version is `MODEL-ONLY`. 
+*MODEL-ONLY* artifacts are shared between all services for the same model version without transformers.
```

Comment:
If you are using transformers, you might want to test different transformers with the same model version and compare results. Otherwise, it doesn't make much sense for now (it does if we had support for setting up custom external k8s clusters, as we discussed once) By default, We always could